### PR TITLE
Df remove value valuelist

### PR DESCRIPTION
**Pull request by dflanderka**
- complete remove of methods value and value_list from Field class and descendants of FieldAlgoBase

### CODE REVIEW

File: unit_tests/fields/field_fe_test.cpp [hunk 1]
```diff
@@ -11,693 +11,1552 @@
...
+        if (ref_shape[0] != shape[0] || ref_shape[1] != shape[1]) {
+            if (ref_shape[0] != shape[0]) std::cout << "Different number of rows of field value and ref value!" << std::endl;
+            if (ref_shape[1] != shape[1]) std::cout << "Different number of cols of field value and ref value!" << std::endl;
+            return false;
+        }
+        double magnitude = std::max( arma::norm(ref_val, 1), arma::norm(field_val, 1) );
+        // abs criterium
+        if (magnitude < 8*std::numeric_limits<double>::epsilon()) return true;
+
+        double error = arma::norm(ref_val - field_val, 1)/magnitude;
```
Comment (jbrezmorf) [inline unit_tests/fields/field_fe_test.cpp:627]: Refactor common code of expect_arma_eqal

File: unit_tests/fields/field_formula_test.cpp [hunk 2]
```diff
@@ -5,192 +5,463 @@
  *      Author: jb
  */
 
-
+#define TEST_USE_MPI
```
Comment (jbrezmorf) [inline unit_tests/fields/field_formula_test.cpp:8]: Consider refactor using common class for various tests.

File: src/fields/field.hh [hunk 3]
```diff
@@ -345,6 +331,9 @@ public:
     /// Implements FieldCommon::value_cache
     const FieldValueCache<double> * value_cache() const override;
 
+    /// Return value_cache
+    FieldValueCache<typename Value::element_type> * field_value_cache();
```
Comment (jbrezmorf) [inline src/fields/field.hh:335]: Is this getter used anywhere?

**Review by jbrezmorf — COMMENTED** [review]
- K diskuzi testy ktere nebyly nahrazeny, asi hlavne ve `field_algo_base_test`.
- Snaha o refaktoring a eliminaci duplicitniho kodu.
- Snaha o lepsi prehlednost co se testuje a schovani technikalii do funkci.

File: src/fields/field.hh (hunk 3 above)
Comment (jbrezmorf) [reply src/fields/field.hh:335]: in field_algo_base_test, define derived class with this getter

File: unit_tests/fields/multi_field_test.cpp [hunk 4]
```diff
@@ -68,176 +83,237 @@ TEST(MultiField, transposition) {
...
 public:
-	typedef MultiField<3, FieldValue<3>::Scalar> ScalarMultiField;
-	typedef ScalarMultiField::SubFieldBaseType ScalarField;
-
-    typedef MultiField<3, FieldValue<3>::VectorFixed> VectorFixedMultiField;
-	typedef VectorFixedMultiField::SubFieldBaseType VectorFixedField;
 
-protected:
-    virtual void SetUp() {
+    class EqData : public FieldSet, public ElementCacheMap {
```
[the remark this answers is not in the document]
Comment (dflanderka) [reply unit_tests/fields/multi_field_test.cpp:90]: New assembly needs more complex tests with full definition of FieldSet (instances of FieldAlgo descendants are not enough). Original check_* functions are replaced by seperate tests for different types of MultiFields. These tests call common methods in test class.

**Pull request merged**